yarn-project/p2p/src/service/data_store.ts

```diff
@@ -0,0 +1,259 @@
+import type { AztecKVStore, AztecMap } from '@aztec/kv-store';
+
+import { type Batch, type Datastore, Key, type KeyQuery, type Pair, type Query } from 'interface-datastore';
+import type { AwaitIterable } from 'interface-store';
+import filter from 'it-filter';
+import map from 'it-map';
```

Comment:
Can we use our own implementation of these helper functions/packages. We should always refrain from importing packages if we don't absolutely need to. It reduces our attack surface considerably.

Reply:
we don't really have iterable utils for these, you think it's worth implementing?